File: practice/exercism/go/hamming/hamming.go
// Package hamming calculates the Hamming distance between two strands of DNA
package hamming

import "github.com/pkg/errors"

// Distance receives two strings (a,b) representing DNA nucleotide sequences, and returns the number of
// nucleotides that differ between the two strands of DNA (Hamming distance).
func Distance(a, b string) (int, error) {

	if len(a) != len(b) {
		return -1, errors.New("Strands must be equal length")
	}

	hd := 0
	for i := range a {
		if a[i] != b[i] {
			hd++
		}
	}

	return hd, nil
}

File: practice/exercism/kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
    fun compute(dna1: String, dna2: String): Int {
        var hdist = 0

        if (dna1.length != dna2.length) {
            throw IllegalArgumentException("left and right strands must be of equal length.")
        }

        for (i in dna1.indices) {
            if (dna1[i] != dna2[i]) hdist += 1
        }

        return hdist
    }
}